hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/file/TestOMRecoverLeaseRequest.java

```diff
@@ -356,6 +359,35 @@ public void testLeaseSoftLimitForHsyncRecoverFile(boolean force) throws Exceptio
     assertNotNull(keyInfo);
   }
 
+  @Test
+  public void testRecoverLeaseWhenOMNotRunning() throws Exception {
+    populateNamespace(true, true, true, true);
+
+    // enable security
+    ozoneManager.getConfiguration().setBoolean(OzoneConfigKeys.OZONE_SECURITY_ENABLED_KEY, true);
+
+    // set OM as Not Running and enable block token
+    when(ozoneManager.isRunning()).thenReturn(false);
+    when(ozoneManager.isGrpcBlockTokenEnabled()).thenReturn(true);
+
+    GenericTestUtils.LogCapturer logCapturer = GenericTestUtils.LogCapturer.captureLogs(OMRecoverLeaseRequest.LOG);
+
+    OMClientResponse omClientResponse = validateAndUpdateCache();
+    OMResponse omResponse = omClientResponse.getOMResponse();
+    assertEquals(OzoneManagerProtocolProtos.Status.OK, omResponse.getStatus());
```

Comment:
is it okay to just let it return OK? Or is it better to return an error? Ideally the client should keep retrying until OM is running.

Reply:
This pass is only for ratis raft log reapply, so skip the token is OK.  But I find a more clean way to handle this. I will upload a new patch.

Reply:
The new patch is tested by Pratyush, and it works, the issue doesn't happen again(It's reproducible) .